CHANGELOG.md

```diff
@@ -19,6 +23,24 @@ All notable changes to this project will be documented in this file.
 
 ---
 
+## [4.0.0](https://github.com/Nike-Inc/Willow/releases/tag/4.0.0)
+
+#### Updated
+- Project and code to Swift 4.0.
+- Small optimization: since the configuration is immutable, we can test the log level before taking the lock or dispatching. Avoids evaluating the log closure if the log will not be written, will increase performance for users with noisy but ignored debug logs.
+- LoggerTests by refactoring into LoggerMessageTests and LoggerMessageStringTests since there are now two separate APIs to test.
+- Example frameworks to not have a default logger instance.
+- Example willow configuration to match new API.
+- Writers such that there are now two protocols: LogWriter and LogModifierWriter. Former is a basic writer. Latter is a writer that also accepts an array of modifiers to apply to incoming messages.
+- All unit tests.
+- Example frameworks to show intended usage of LogMessages.
+
+#### Added
+- Added support for structured messages and simplified Logger setup.
+
+#### Removed
+- Configuration from Willow. Callers now pass a log level, writers, and execution method to init(). Greatly simplifies setup and usage.
+ 
```

Comment:
We should take another pass on these after the PR is merged to add attribution and PR number. Doesn't need to be in this PR.